### PR TITLE
[20644] Add 2.14.x to versions.rst

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR adds 2.14.x to the list of supported versions.

Related implementation PR:

* eProsima/Fast-DDS#4579

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- _N/A_: Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- _N/A_: Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

File: docs/notes/versions.rst [hunk 1]
```diff
@@ -49,6 +49,61 @@ The following table shows the minimum version required of the Fast DDS build sys
...
+              - Tier 3: GCC 9
+            * - MacOS Mojave (10.14)
+              - Tier 1: Clang 15
+              - **───**
+              - **───**
+            * - Windows 10
+              - Tier 1: MSVC v142 (Visual Studio 2019) |br|
+                Tier 2: MSVC v141 (Visual Studio 2017)
+              - Tier 2: MSVC v142 (Visual Studio 2019), |br|
+                Tier 2: MSVC v141 (Visual Studio 2017)
```
Comment (JesusPoderoso) [inline docs/notes/versions.rst:84]: It seems like someone left a comma in one of the cases but not in the other.
I would suggest removing the comma in all cases, not just 2.14.x

```suggestion
              - Tier 1: MSVC v142 (Visual Studio 2019) |br|
                Tier 2: MSVC v141 (Visual Studio 2017)
              - Tier 2: MSVC v142 (Visual Studio 2019) |br|
                Tier 2: MSVC v141 (Visual Studio 2017)
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Little NIT

**Comment by EduPonz** [comment]
@richiprosima please test docs

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**